core/config/toml/types.go

```diff
@@ -715,6 +754,110 @@ func (w *WebServerTLS) setFrom(f *WebServerTLS) {
 	}
 }
 
+type WebServerLDAP struct {
+	ServerTLS                   *bool
+	SessionTimeout              *models.Duration
+	QueryTimeout                *models.Duration
+	BaseUserAttr                *string
+	BaseDN                      *string
+	UsersDN                     *string
+	GroupsDN                    *string
+	ActiveAttribute             *string
+	ActiveAttributeAllowedValue *string
+	AdminUserGroupCN            *string
+	EditUserGroupCN             *string
+	RunUserGroupCN              *string
+	ReadUserGroupCN             *string
+	UserApiTokenEnabled         *bool
+	UserAPITokenDuration        *models.Duration
+	UpstreamSyncInterval        *models.Duration
+	UpstreamSyncRateLimit       *models.Duration
+}
+
+func (w *WebServerLDAP) setFrom(f *WebServerLDAP) {
+	if v := f.ServerTLS; v != nil {
```

Comment:
This is a pattern used since I introduced TOML config last year, so out of scope for this particular PR since it's so widely used, but we should definiley consider this. I don't recall if there were any irregularities that required these manual implementations.

Reply:
Oh we do need some way of indicating how deep to crawl, because we only want to descend through our own types, not inside of the types for individual TOML fields. We don't always use struct tags, so that won't work. Maybe we would need to match the toml decoder's logic for TextUnmarshaler() :thinking: